src/lib.rs

```diff
@@ -226,12 +226,45 @@
 
 use std::collections::HashMap;
 use std::env::VarError;
+use std::fmt::Debug;
 use std::str::FromStr;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::{Arc, Condvar, Mutex, MutexGuard, RwLock, TryLockError};
 use std::time::{Duration, Instant};
 use std::{env, thread};
 
+type Callback = Box<dyn Fn() + Send + Sync>;
+
+#[derive(Clone)]
+struct SyncCallback {
+    callback: Arc<Callback>,
```

Comment:
can we do `Arc<dyn Fn() + Send + Sync>`?

Reply:
Of course we can. I just to make this code shorter.